src/lib/utils/redisCache.ts

```diff
@@ -0,0 +1,44 @@
+/* eslint-disable no-console */
+import { DataCache } from 'next-drupal'
+import { createClient, RedisClientType } from 'redis'
+
+export async function createRedisClient(url): Promise<RedisClientType> {
+  let isReady = false
+
+  let redisClient: RedisClientType
+  if (isReady === false) {
+    redisClient = createClient({ url })
+    redisClient.on('error', (err) => console.error(`Redis Error: ${err}`))
+    redisClient.on('connect', () => console.info('Redis connected'))
+    redisClient.on('reconnecting', () => console.info('Redis reconnecting'))
+    redisClient.on('ready', () => {
+      isReady = true
+      console.info('Redis ready!')
+    })
+  }
+
+  await redisClient.connect()
```

Comment:
I find I generally have to think about `Promise`/`async`/`await` stuff a million times to chew on it, but I think we can likely do away with the `await` here, as well as the `async` in the function signature. `connect` will return a `Promise`, which, to me, in this case, is more clear than awaiting something within the function itself. I think we can just return the result of the `.connect()` call, which is effectively saying that we're returning something that has attempted to make a connection and will eventually be a connection. Then, we can `await` that when we need to call a method on it in the `redisCache` function below.

This might just be my flavor of reasoning about this, so take it with a grain of salt.